src/utils.py

```diff
@@ -252,8 +254,11 @@ def convert_to_df(data: Dict[str, Any]) -> pd.DataFrame:
     data = convert_to_list(data)
     data = flatten_dict(data)
     lens = [len(x) for x in data.values()]
-    assert min(lens) == max(lens)  # all lengths are equal!
-    # NOTE: pandas can't haneld high dimensional np arrays, so we just use lists
+    # all lengths are equal!
+    if not min(lens) == max(lens):
```

Comment:
Not sure what the benefit of this is over a simple assert. Requires users to have the `ipdb` module loaded and will issue a nasty `ModuleNotFoundError` which might confuse the users.